tests/component/records/pandas/test_to_csv_options.py

```diff
@@ -0,0 +1,236 @@
+            'YYYY-MM-DD HH:MI:SS': '%Y-%m-%d %H:%M:%S.%f',
+            'YYYY-MM-DD HH24:MI:SSOF': '%Y-%m-%d %H:%M:%S.%f%z',
+            # TODO: Why are we including seconds here are at all?
+            'MM/DD/YY HH24:MI': '%m/%d/%y %H:%M:%S.%f',
+        }
+        compatible_dateformat = {
+            'YYYY-MM-DD HH:MI:SSOF': 'YYYY-MM-DD',
+            'YYYY-MM-DD HH:MI:SS': 'YYYY-MM-DD',
+            'YYYY-MM-DD HH24:MI:SSOF': 'YYYY-MM-DD',
+            'MM/DD/YY HH24:MI': 'MM/DD/YY',
+        }
+        for datetimeformattz in DATETIMETZ_CASES:
+            records_format = DelimitedRecordsFormat(hints={
+                # Pandas doesn't consider dateformats to be separate
+                # from datetime/datetimetz formats, so they need to be
+                # consistent
+                'dateformat': compatible_dateformat[datetimeformattz],
+                'datetimeformat': datetimeformattz.replace('OF', ''),
+                'datetimeformattz': datetimeformattz,
+                'compression': None,
+            })
+            unhandled_hints = set(records_format.hints)
+            processing_instructions = ProcessingInstructions()
+            try:
+                options = pandas_to_csv_options(records_format,
+                                                unhandled_hints,
+                                                processing_instructions)
+            except NotImplementedError:
+                if datetimeformattz in known_failures:
+                    continue
+                else:
+                    raise
+            self.assertEqual(options['date_format'], expectations[datetimeformattz])
+
+            fileobj = io.StringIO(create_sample(datetimeformattz))
+            df = pandas.DataFrame(data={'datetime':
+                                        [pandas.Timestamp(day=SAMPLE_DAY,
+                                                          month=SAMPLE_MONTH,
+                                                          year=SAMPLE_YEAR,
+                                                          hour=SAMPLE_HOUR,
+                                                          minute=SAMPLE_MINUTE,
+                                                          second=SAMPLE_SECOND)]},
+                                  columns=['datetime'])
+            df.to_csv(path_or_buf=fileobj,
+                      index=False,
+                      **options)
+            output = fileobj.getvalue()
+            # In reality this isn't used raw, as Pandas doesn't really
+            # try to handle lone dates or times.  Instead, we use
+            # prep_for_csv() to preconvert these Serieses into strings.
+
+            # Pandas drops the timezone offset if it's zero, which it
+            # is in our example
+            datetimeformattz_minus_offset = datetimeformattz.replace('OF', '')
+            sample = create_sample(datetimeformattz_minus_offset)
+            # TODO: Why are we including microsecond?  Is that in
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker